graphql/src/execution/execution.rs

```diff
@@ -556,58 +570,27 @@ pub fn collect_fields<'a>(
-
-                    // Create a field group for this response key on demand and
-                    // append the selection field to this group.
-                    grouped_fields.entry(response_key).or_default().push(field);
+                    output.entry(response_key).or_default().push(field);
                 }
 
                 q::Selection::FragmentSpread(spread) => {
                     // Only consider the fragment if it hasn't already been included,
                     // as would be the case if the same fragment spread ...Foo appeared
                     // twice in the same selection set
-                    if !visited_fragments.contains(&spread.fragment_name) {
-                        visited_fragments.insert(&spread.fragment_name);
-
-                        // Resolve the fragment using its name and, if it applies, collect
-                        // fields for the fragment and group them
-                        ctx.query
-                            .get_fragment(&spread.fragment_name)
-                            .and_then(|fragment| {
-                                // We have a fragment, only pass it on if it applies to the
-                                // current object type
-                                if does_fragment_type_apply(
-                                    ctx,
-                                    object_type,
-                                    &fragment.type_condition,
-                                ) {
-                                    Some(fragment)
-                                } else {
-                                    None
-                                }
-                            })
-                            .map(|fragment| {
-                                // We have a fragment that applies to the current object type,
-                                // collect its fields into response key groups
-                                let fragment_grouped_field_set = collect_fields(
-                                    ctx,
-                                    object_type,
-                                    iter::once(&fragment.selection_set),
-                                    Some(visited_fragments.clone()),
-                                );
-
-                                // Add all items from each fragments group to the field group
-                                // with the corresponding response key
-                                for (response_key, mut fragment_group) in fragment_grouped_field_set
-                                {
-                                    grouped_fields
-                                        .entry(response_key)
-                                        .or_default()
-                                        .append(&mut fragment_group);
-                                }
-                            });
+                    if visited_fragments.insert(&spread.fragment_name) {
```

Comment:
Interesting observation. If we were to check for this, it would be checked in a validation step and not here. But I don't see any reason to bother with not supporting cycles when they can be supported just fine. I've added a note here.